web/src/redux/apis/TraceTest.api.ts

```diff
@@ -88,11 +93,35 @@ const TraceTestAPI = createApi({
       transformResponse: (rawTestResultList: TRawTestRun[]) =>
         rawTestResultList.map(rawTestResult => TestRun(rawTestResult)),
     }),
+
     getRunById: build.query<TTestRun, {runId: string; testId: string}>({
       query: ({testId, runId}) => `/tests/${testId}/run/${runId}`,
       providesTags: result => (result ? [{type: Tags.TEST_RUN, id: result?.id}] : []),
       transformResponse: (rawTestResult: TRawTestRun) => TestRun(rawTestResult),
+      async onCacheEntryAdded(arg, {cacheDataLoaded, cacheEntryRemoved, updateCachedData}) {
+        try {
```

Comment:
I think this is where the magic kind of happens right? I'm not sure how different we could handle this, maybe we could start thinking about splitting up the endpoints from the main API file so this doesn´t get that big.